src/libraries/Common/src/System/Collections/Generic/BitHelper.cs

```diff
@@ -19,19 +21,22 @@ internal BitHelper(Span<int> span, bool clear)
 
         internal void MarkBit(int bitPosition)
         {
-            int bitArrayIndex = bitPosition / IntSize;
-            if ((uint)bitArrayIndex < (uint)_span.Length)
+            (uint bitArrayIndex, uint position) = Math.DivRem((uint)bitPosition, IntSize);
```

Comment:
Is the code with DivRem as efficient as it would be if this used unsigned div and mod directly?

DivRem is not treated as the intrinsic yet. My guess is that some unnecessary instructions compared to just using unsigned div and mod directly.

Reply:
🎉 this gives really nice codegen. Thanks!

The jit-diff from this change allone (based on da31c0b334046205da7aa776dceaecc2c82e56ba to d9f4e01930f78f1c42c29514ea18fe05dee248f6) shows good improvements.

```
Top method improvements (bytes):
        -118 (-12,14 % of base) : System.Private.CoreLib.dasm - System.Collections.Generic.HashSet`1:SymmetricExceptWithEnumerable(System.Collections.Generic.IEnumerable`1[System.__Canon]):this
         -84 (-12,65 % of base) : System.Private.CoreLib.dasm - System.Collections.Generic.HashSet`1:IntersectWithEnumerable(System.Collections.Generic.IEnumerable`1[System.__Canon]):this
         -30 (-3,69 % of base) : System.Private.CoreLib.dasm - System.Collections.Generic.HashSet`1:CheckUniqueAndUnfoundElements(System.Collections.Generic.IEnumerable`1[System.__Canon],bool):System.ValueTuple`2[System.Int32, System.Int32]:this
         -10 (-22,73 % of base) : System.Private.CoreLib.dasm - System.Collections.Generic.BitHelper:IsMarked(int):bool:this
         -10 (-21,74 % of base) : System.Private.CoreLib.dasm - System.Collections.Generic.BitHelper:MarkBit(int):this
```

I tried without the local Span, but then there were some regressions in larger methods.
In SharpLab, etc. no bound check was shown for the method alone, but in the jit-diff I saw some, so brought the local Span back.